--- origsrc/mate-notification-daemon-1.14.0/src/daemon/daemon.c	2015-12-14 14:36:59.000000000 -0600
+++ src/mate-notification-daemon-1.14.0/src/daemon/daemon.c	2016-05-22 15:26:20.240840600 -0500
@@ -99,7 +99,6 @@ typedef struct {
 struct _NotifyDaemonPrivate {
 	guint next_id;
 	guint timeout_source;
-	guint exit_timeout_source;
 	GHashTable* idle_reposition_notify_ids;
 	GHashTable* monitored_window_hash;
 	GHashTable* notification_hash;
@@ -154,33 +153,6 @@ static void _notify_timeout_destroy(Noti
 	g_free(nt);
 }
 
-static gboolean do_exit(gpointer user_data)
-{
-	exit(0);
-	return FALSE;
-}
-
-static void add_exit_timeout(NotifyDaemon* daemon)
-{
-	g_assert (daemon != NULL);
-
-	if (daemon->priv->exit_timeout_source > 0)
-		return;
-
-	daemon->priv->exit_timeout_source = g_timeout_add_seconds(IDLE_SECONDS, do_exit, NULL);
-}
-
-static void remove_exit_timeout(NotifyDaemon* daemon)
-{
-	g_assert (daemon != NULL);
-
-	if (daemon->priv->exit_timeout_source == 0)
-		return;
-
-	g_source_remove(daemon->priv->exit_timeout_source);
-	daemon->priv->exit_timeout_source = 0;
-}
-
 static void create_stack_for_monitor(NotifyDaemon* daemon, GdkScreen* screen, int monitor_num)
 {
 #if GTK_CHECK_VERSION (3, 8, 0)
@@ -412,8 +384,6 @@ static void notify_daemon_init(NotifyDae
 	daemon->priv->next_id = 1;
 	daemon->priv->timeout_source = 0;
 
-	add_exit_timeout(daemon);
-
 	daemon->gsettings = g_settings_new (GSETTINGS_SCHEMA);
 
 	g_signal_connect (daemon->gsettings, "changed::" GSETTINGS_KEY_POPUP_LOCATION, G_CALLBACK (on_popup_location_changed), daemon);
@@ -510,8 +480,6 @@ static void notify_daemon_finalize(GObje
 		gdk_window_remove_filter(NULL, (GdkFilterFunc) _notify_x11_filter, daemon);
 	}
 
-	remove_exit_timeout(daemon);
-
 	g_hash_table_destroy(daemon->priv->monitored_window_hash);
 	g_hash_table_destroy(daemon->priv->idle_reposition_notify_ids);
 	g_hash_table_destroy(daemon->priv->notification_hash);
@@ -613,11 +581,6 @@ static void _close_notification(NotifyDa
 		}
 
 		g_hash_table_remove(priv->notification_hash, &id);
-
-		if (g_hash_table_size(daemon->priv->notification_hash) == 0)
-		{
-			add_exit_timeout(daemon);
-		}
 	}
 }
 
@@ -835,11 +798,6 @@ static gboolean _check_expiration(Notify
 	if (!has_more_timeouts)
 	{
 		daemon->priv->timeout_source = 0;
-
-		if (g_hash_table_size (daemon->priv->notification_hash) == 0)
-		{
-			add_exit_timeout(daemon);
-		}
 	}
 
 	return has_more_timeouts;
@@ -917,7 +875,6 @@ static NotifyTimeout* _store_notificatio
 	_calculate_timeout(daemon, nt, timeout);
 
 	g_hash_table_insert(priv->notification_hash, g_memdup(&id, sizeof(guint)), nt);
-	remove_exit_timeout(daemon);
 
 	return nt;
 }
