--- origsrc/qoauth-1.0.1-src/src/src.pro	2010-08-01 09:08:59.000000000 -0500
+++ src/qoauth-1.0.1-src/src/src.pro	2013-08-28 21:13:37.262260400 -0500
@@ -61,7 +61,7 @@ else:unix {
     system( ./pcfile.sh $${INSTALL_PREFIX} $${VERSION} )
     pkgconfig.files = qoauth.pc
     
-    contains(QMAKE_HOST.arch, x86_64) {
+    contains(QMAKE_HOST.arch, x86_64):exists($${INSTALL_PREFIX}/lib64) {
       target.path = $${INSTALL_PREFIX}/lib64
     } else {
       target.path = $${INSTALL_PREFIX}/lib
